RepositoryCompiler/CodeModel/CodeParsers/CSharp/CaDETClassMetricCalculator.cs

```diff
@@ -123,5 +124,43 @@ private int GetNumberOfMethodsDeclared(CaDETClass parsedClass)
         {
             return parsedClass.Members.Count(method => method.Type.Equals(CaDETMemberType.Method));
         }
+
+        /// <summary>
+        /// CC - Class cohesion
+        /// DOI: 10.14257/ijmue.2014.9.6.35
+        /// </summary>
+        private double? GetCCClassCohesion(CaDETClass parsedClass)
+        {
+            IEnumerable<CaDETMember> memberMethods = parsedClass.Members.Where(m => m.Type != CaDETMemberType.Property);
+            IEnumerable<CaDETMember> memberProperties = parsedClass.Members.Where(m => m.Type == CaDETMemberType.Property);
+            double fieldsCount = parsedClass.Fields.Count + memberProperties.Count();
+
+            if (fieldsCount == 0 || memberMethods.Count() == 0)
```

Comment:
What happens if fieldCount is zero, but we have field defining properties (e.g., string SomeProp {get;set;})?

Reply:
fieldsCount is number of all fields and members that are properties - defined in 136. Maybe renaming the variable to "DataMembersCount" like you suggested bellow would make this more clear?